packages/node/src/utils.ts

```diff
@@ -79,3 +81,27 @@ export function isEdgeRuntime(runtime?: string): runtime is EdgeRuntimes {
     Object.values(EdgeRuntimes).includes(runtime as EdgeRuntimes)
   );
 }
+
+export function detectServerlessLauncherType(
+  entrypoint: string,
+  targetRuntime: string
+) {
+  const launcherType = hasEdgeSignature(entrypoint) ? 'edge-light' : 'Nodejs';
+  checkLauncherCompatibility(entrypoint, launcherType, targetRuntime);
+  return launcherType;
+}
+
+function checkLauncherCompatibility(
+  entrypoint: string,
+  launcherType: LauncherConfiguration['launcherType'],
+  runtime: string
+) {
+  if (launcherType === 'edge-light') {
+    const nodeMajor = Number(runtime.replace('nodejs', '').split('.')[0]);
+    if (nodeMajor < 18) {
+      throw new Error(
+        `${entrypoint}: ${launcherType} launcher type can only be used with node.js 18 and later`
```

Comment:
Is this an error message that a user will see at build time? Seems like we need to use `NowBuilderError()` and add a link to the documentation. Example:

https://github.com/vercel/vercel/blob/5b88f673f89c44b9e91e44247e29be6bed43805a/packages/build-utils/src/fs/node-version.ts#L64-L70

Reply:
Yes, good call! I have a similar error in my build-container PR.